SEED-platform/seed
# Add support for SMTP email backend to Docker configuration

#### Any background context you want to provide?
The Docker image currently only supports the AWS SES
email backend. We (OPEN) would like the flexibility to use
SMTP for sending email.

#### What's this PR do?
This PR changes the Docker config file so that
it allows configuring different values for the email backend
that Django will use. It also copies the values from
Django's SMTP-related environment variables into memory.
(See the [Django docs for the full list](https://docs.djangoproject.com/en/3.0/topics/email/#smtp-backend) of available configuration variables.)

#### How should this be manually tested?
1. Build a new version of the Docker image.
2. Customize the docker-compose config file
    to include values for the following environment variables.
  - `DJANGO_EMAIL_BACKEND=django.core.mail.backends.smtp.EmailBackend`
  - `SERVER_EMAIL=seed_server@example.com`
  - `EMAIL_HOST=smtp.example.com`
  - `EMAIL_PORT=587`
  - `EMAIL_USE_TLS=True` _or_ `EMAIL_USE_SSL=True`
  - `EMAIL_HOST_USER=seed_account_support@example.com`
  - `EMAIL_HOST_PASSWORD=password123`
3. Run the Docker image using docker-compose.
4. Log in to the Web application.
5. Create or invite a new user with a real email address to join an organization
    in SEED.
6.  Verify that the user receives an email about their new account or inviting them to a new organization.
